src/integer.rs

```diff
@@ -438,6 +443,36 @@ impl Integer {
             IntegerType::Bignum(big) => big.to_usize(),
         }
     }
+
+    /// Normalize `self`. If `self` is a `Fixnum`, returns `self`. If `self` is
+    /// a `Bignum`, if it is small enough to fit in a `Fixnum`, returns a
+    /// `Fixnum` with the same value. Otherwise, returns `self`.
+    pub fn norm(&self) -> Self {
+        match self.integer_type() {
+            IntegerType::Fixnum(_) => *self,
+            IntegerType::Bignum(big) => unsafe {
+                Integer::from_rb_value_unchecked(rb_big_norm(big.as_rb_value()))
+            },
+        }
+    }
+
+    fn binary_operation_visit<T>(
+        &self,
+        other: &Self,
+        rust_op: fn(Fixnum, Fixnum) -> T,
+        ruby_op: fn(rb_sys::VALUE, rb_sys::VALUE) -> T,
```

Comment:
Could you import all the `rb_sys` types/functions up top and use the names unqualified please. Just to match the rest of the project.